utils/clean.py

```diff
@@ -97,5 +104,215 @@ def clean_state(self) -> (pd.DataFrame, pd.DataFrame, pd.DataFrame):
+
+        # details["company"] = employer
+
+        cleaned_transactions, cleaned_details = ArizonaCleaner.clean(
+            [transactions, details]
+        )
+
+        standardized_transactions, standardized_details = ArizonaCleaner.standardize(
+            [cleaned_transactions, cleaned_details]
+        )
+
+        (
+            az_transactions,
+            az_individuals,
+            az_organizations,
+        ) = ArizonaCleaner.create_tables(
+            [standardized_transactions, standardized_details]
+        )
+
+        return (az_transactions, az_individuals, az_organizations)
+
+    def create_tables(
+        data: list[pd.DataFrame],
+    ) -> (pd.DataFrame, pd.DataFrame, pd.DataFrame):
+        """split up cleaned and standardized tables to undergo final formatting
+
+        We split up the details tables into individuals (individual contributors
+        and donors) and organizations (everything else), which then undergo
+        conversion into fully schema-compatible transactions, individual details
+        and organization details tables.
+
+        args: cleaned and standardized transactions and details tables
+
+        returns: three tables for transactions,
+        individual details, and organization details
+
+        """
+
+        transactions, details = data[0], data[1]
+
+        individual_details = details[
+            (details["entity_type"] == "Individual")
+            | (details["entity_type"] == "Candidate")
+        ]
+        organization_details = details[
+            (details["entity_type"] != "Individual")
+            & (details["entity_type"] != "Candidate")
+        ]
+
+        # individual_details = cs_details[
+        #     cs_details["entity_type"] == ("Individual Contributors" or "Candidate")
+        # ]
+
+        # organization_details = cs_details[
+        #     cs_details["entity_type"] != ("Individual Contributors" or "Candidate")
+        # ]
+
+        az_transactions = az_transactions_convert(transactions)
+
+        if len(individual_details) > 0:
+            az_individuals = az_individuals_convert(transactions, individual_details)
+        else:
+            az_individuals = None
+
+        if len(organization_details) > 0:
+            az_organizations = az_organizations_convert(organization_details)
+        else:
+            az_organizations = None
+
+        return (az_transactions, az_individuals, az_organizations)
+
+    def standardize(details_df_list: list[pd.DataFrame]) -> list[pd.DataFrame]:
+        """standardize names of entities
+
+        takes in details dataframe and replaces the names of
+        organization types to fit into the schema when appropriate
+
+        args: details dataframe
+
+        returns: details dataframe with relevant entity type
+        names replaced by those for the regular schema
+        """
+
+        transactions_df, details_df = details_df_list[0], details_df_list[1]
+
+        az_entity_name_dictionary = {
```

Comment:
should be defined above instead of within the function see [here](https://stackoverflow.com/questions/5960337/how-to-create-abstract-properties-in-python-abstract-classes)